### ISSUE TITLE
[Documentation]: Combine "contributing.md" and "development.md"

### ISSUE THREAD

**Issue by neubig** [OpenDevin/OpenDevin#2386]
**What problem or use case are you trying to solve?**

We currently have two documents on how to contribute to OpenDevin, but it's confusing about which one to look at:
* https://github.com/OpenDevin/OpenDevin/blob/main/CONTRIBUTING.md
* https://github.com/OpenDevin/OpenDevin/blob/main/Development.md

We should consider combining the content of the two.

**Comment by tobitege** [comment]
Might actually be good to keep them separate, but rather improve upon their content, which doesn't necessarily have a big overlap (dev <-> enduser):
https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors#adding-a-contributing-file

**Comment by neubig** [comment]
I do think that this would be a possibility, but (1) right now the overlap between these two files seems pretty high (both deal with environment setup for devs, etc), and (2) the files are not super-long. Because of this, I think it might be nice to just have all information in one place but with a table of contents so you can quickly jump to the relevant information for now, and if the file length gets too unwieldy we could consider splitting them again.

**Comment by edwardbrazier** [comment]
Thanks for raising this.
I would be keen to at least have clearer references between them:
When I read contributing.md, I didn't know that development.md existed and I just tried to guess about how to build the program, until neubig pointed me to the right one.

**Comment by mamoodi** [comment]
@neubig When I first picked this issue up, I tried combining the two and it didn't look great tbh. It became rather large so I fixed it by referencing each doc in the beginning of the other doc. 

There isn't really overlap between the two and I tried to explain the use case of each document. Take a look at the new files and reopen with any concerns if you feel like it did not resolve the main problem.
